nmigen/vendor/tinyfpga_b.py

```diff
@@ -0,0 +1,34 @@
+from ..build import *
+from .fpga.lattice_ice40 import LatticeICE40Platform, TinyProgrammerMixin
+
+
+__all__ = ["TinyFPGABPlatform"]
+
+
+class TinyFPGABPlatform(TinyProgrammerMixin, LatticeICE40Platform):
+    device    = "lp8k"
+    package   = "cm81"
+    clocks    = [
+        ("clk16", 16e6),
+    ]
+    resources = [
+        Resource("clk16", 0, Pins("B2", dir="i"), extras=["IO_STANDARD=LVCMOS33"]),
+
+        Resource("user_led", 0, Pins("B3", dir="o"), extras=["IO_STANDARD=LVCMOS33"]),
+
+        Resource("usb", 0,
+            Subsignal("d_p", Pins("B4", dir="io")),
+            Subsignal("d_n", Pins("A4", dir="io")),
+            Subsignal("pull_up", Pins("A3", dir="o")),
+            extras=["IO_STANDARD=SB_LVCMOS33"]
+        ),
+
+        Resource("spiflash", 0,
+            Subsignal("cs_n", Pins("F7", dir="o")),
+            Subsignal("clk",  Pins("G7", dir="o")),
+            Subsignal("mosi", Pins("G6", dir="io")),
+            Subsignal("miso", Pins("H7", dir="io")),
+            extras=["IO_STANDARD=SB_LVCMOS33"]
+        ),
+    ]
+    prog_mode = "flash"
```

Comment:
The TinyFPGA programmer always uses the flash mode, so it doesn't need this attribute. (It's used only in `IceStormProgrammerMixin`, as you can see.)